### PR TITLE
Getting Started docs: Fetching and Streaming

### PR DESCRIPTION
**Pull request by delbaoliveira**
Closes: https://linear.app/vercel/issue/DOC-3699/page-fetching-and-streaming-data

### CODE REVIEW

File: docs/01-app/01-getting-started/06-data-fetching-and-streaming.mdx [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+### Client Components
+
+To fetch data in Client Components, we recommend using a React library like [SWR](https://swr.vercel.app/) or [React Query](https://tanstack.com/query/latest). For example:
+
+```tsx filename="app/blog/page.tsx" switcher
+'use client'
+
+import useSWR from 'swr'
+import { fetcher } from '@/lib/fetcher'
```
Comment (molebox) [inline docs/01-app/01-getting-started/06-data-fetching-and-streaming.mdx:96]: Can we give a small example of what this `fetcher` function is/does?

**Review by molebox — COMMENTED** [review]
Left some small comments

File: docs/01-app/01-getting-started/06-data-fetching-and-streaming.mdx (hunk 1 above)
Comment (delbaoliveira) [reply docs/01-app/01-getting-started/06-data-fetching-and-streaming.mdx:96]: Yes, can do! It's something you set up with SWR.

File: docs/01-app/01-getting-started/06-data-fetching-and-streaming.mdx [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+        <li key={post.id}>{post.title}</li>
+      ))}
+    </ul>
+  )
+}
+```
+
+### Client Components
+
+To fetch data in Client Components, we recommend using a React library like [SWR](https://swr.vercel.app/) or [React Query](https://tanstack.com/query/latest). For example:
```
Comment (leerob) [inline docs/01-app/01-getting-started/06-data-fetching-and-streaming.mdx:90]: Let's teach the `use()` pattern?

File: docs/01-app/01-getting-started/06-data-fetching-and-streaming.mdx [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+    </ul>
+  )
+}
+```
+
+## Streaming
+
+> **Warning:** The content below assumes the [`dynamicIO` config option](/docs/app/api-reference/config/next-config-js/dynamicIO) is enabled in your application. The flag was introduced in Next.js 15 canary.
+
+When using `async/await` in Server Components, Next.js will opt into **dynamic rendering**. This means the data will be fetched and rendered on the server for every user request. If there are any slow data requests, the whole route will be blocked from rendering. To improve the initial load time and user experience, you can use streaming to break up the page's HTML into smaller chunks and progressively send those chunks from the server to the client.
```
Comment (leerob) [inline docs/01-app/01-getting-started/06-data-fetching-and-streaming.mdx:148]: Should we clarify that Next.js streams by default?

File: docs/01-app/01-getting-started/06-data-fetching-and-streaming.mdx (hunk 2 above)
Comment (delbaoliveira) [reply docs/01-app/01-getting-started/06-data-fetching-and-streaming.mdx:90]: Good shout, I'll add an example.

File: docs/01-app/01-getting-started/06-data-fetching-and-streaming.mdx (hunk 3 above)
Comment (delbaoliveira) [reply docs/01-app/01-getting-started/06-data-fetching-and-streaming.mdx:148]: Sorry, what do you mean?

Comment (leerob) [reply docs/01-app/01-getting-started/06-data-fetching-and-streaming.mdx]: This paragraph reads like you need to opt-into streaming. But the App Router will stream by default, it's just how much content is included in the initial payload depends on your suspense boundaries. That's my understanding (which could be wrong!). Basically what I'm saying it, you can't turn streaming _off_, and you don't need to enable it.

**Review by leerob — APPROVED** [review]

**Pull request merged**